### ISSUE TITLE
Evaluation got low accuracy using pretrained model 

### ISSUE THREAD

**Issue by qunluo** [NervanaSystems/distiller#265]
Hello guys, I run the command in examples/quantization/post_train_quant/command_line.md, but I got low accuracy.
The log is as follow:
2019-05-22 18:09:06,187 - Log file for this run: /data/lcm/distiller/examples/classifier_compression/logs/2019.05.22-180906/2019.05.22-180906.log
2019-05-22 18:09:06,260 - Number of CPUs: 72
2019-05-22 18:09:11,468 - Number of GPUs: 2
2019-05-22 18:09:11,469 - CUDA version: 9.0.176
2019-05-22 18:09:11,469 - CUDNN version: 7402
2019-05-22 18:09:11,469 - Kernel: 3.13.0-93-generic
2019-05-22 18:09:11,470 - Python: 3.5.3 (default, Apr 22 2017, 00:00:00)
[GCC 4.8.4]
2019-05-22 18:09:11,470 - PyTorch: 1.0.1
2019-05-22 18:09:11,470 - Numpy: 1.16.3
2019-05-22 18:09:12,012 - Active Git branch: master
2019-05-22 18:09:12,029 - Git commit: 3153687337ec7ef4cb717caeda86bee3545215fa
2019-05-22 18:09:12,029 - Command line: compress_classifier.py -a resnet50 --pretrained /data/lcm/data/imagenet/ --evaluate
2019-05-22 18:09:12,937 - => using pretrained resnet50 model for ImageNet
2019-05-22 18:09:18,216 - Optimizer Type: <class 'torch.optim.sgd.SGD'>
2019-05-22 18:09:18,217 - Optimizer Args: {'dampening': 0, 'momentum': 0.9, 'weight_decay': 0.0001, 'nesterov': False, 'lr': 0.1}
2019-05-22 18:09:55,153 - Dataset sizes:
  training=1153051
  validation=128116
  test=50000
2019-05-22 18:09:55,153 - --- test ---------------------
2019-05-22 18:09:55,153 - 50000 samples (256 per mini-batch)
2019-05-22 18:10:19,660 - Test: [   10/  195]    Loss 18.425109    Top1 0.039062    Top5 0.117188
2019-05-22 18:10:36,376 - Test: [   20/  195]    Loss 18.377566    Top1 0.039062    Top5 0.253906
2019-05-22 18:10:59,612 - Test: [   30/  195]    Loss 18.375945    Top1 0.065104    Top5 0.260417
2019-05-22 18:11:18,283 - Test: [   40/  195]    Loss 18.309645    Top1 0.058594    Top5 0.273437
2019-05-22 18:11:39,780 - Test: [   50/  195]    Loss 18.326377    Top1 0.054687    Top5 0.273437
2019-05-22 18:11:59,250 - Test: [   60/  195]    Loss 18.328863    Top1 0.065104    Top5 0.299479
2019-05-22 18:12:21,389 - Test: [   70/  195]    Loss 18.345041    Top1 0.094866    Top5 0.329241
2019-05-22 18:12:41,015 - Test: [   80/  195]    Loss 18.349581    Top1 0.112305    Top5 0.327148
2019-05-22 18:13:03,212 - Test: [   90/  195]    Loss 18.334290    Top1 0.108507    Top5 0.338542
2019-05-22 18:13:22,950 - Test: [  100/  195]    Loss 18.345718    Top1 0.113281    Top5 0.332031
2019-05-22 18:13:45,083 - Test: [  110/  195]    Loss 18.349431    Top1 0.110085    Top5 0.326705
2019-05-22 18:14:03,126 - Test: [  120/  195]    Loss 18.342557    Top1 0.113932    Top5 0.325521
2019-05-22 18:14:23,808 - Test: [  130/  195]    Loss 18.349034    Top1 0.105168    Top5 0.312500
2019-05-22 18:14:35,845 - Test: [  140/  195]    Loss 18.345225    Top1 0.100446    Top5 0.315290
2019-05-22 18:14:46,186 - Test: [  150/  195]    Loss 18.349084    Top1 0.096354    Top5 0.304688
2019-05-22 18:14:56,403 - Test: [  160/  195]    Loss 18.355142    Top1 0.095215    Top5 0.302734
2019-05-22 18:15:06,654 - Test: [  170/  195]    Loss 18.350308    Top1 0.089614    Top5 0.298713
2019-05-22 18:15:17,078 - Test: [  180/  195]    Loss 18.348971    Top1 0.091146    Top5 0.303819
2019-05-22 18:15:27,297 - Test: [  190/  195]    Loss 18.346163    Top1 0.088405    Top5 0.306332
2019-05-22 18:15:32,740 - ==> Top1: 0.090    Top5: 0.308    Loss: 18.340

According to examples/quantization/post_train_quant/command_line.md, the accuracy should be top1:76.13%.

|   | Mode       | # Bits Acts | # Bits Weights | Per-Channel | Clip Acts             | Top-1 Accuracy |
|---|------------|-------------|----------------|-------------|-----------------------|----------------|
| 1 | FP32       | 32          | 32             | N/A         |                       | 76.13%

**Comment by guyjacob** [comment]
This is really as simple a command-line as it gets, so not sure what could be going on. It seems like you're getting random results, so perhaps the pre-trained model isn't being loaded for some reason. Is this the first time you're running it? If so, and you haven't used torchvision, then it should download the model from the PyTorch model zoo to your machine. The default save path for it is `~/.torch/models`. Can you check if there is something there? The file name for the ResNet-50 model specifically is `resnet50-19c8e357.pth`. Maybe something went wrong with the download (although I'd expect some error to be raised...).

**Comment by nzmora** [comment]
Hi @qunluo, 

Another option is that your ImageNet dataset is incorrect.  A totally random Top1 (i.e. guesses produced by a uniform distribution) for classification with 1000 classes is 1/1000 = 0.1.  You are getting Top1 = 0.09.  So either the DS  has a problem (e.g. labels do not match the image classes), or as @guyjacob said, the model is not trained.
Cheers
Neta

**Comment by qunluo** [comment]
Hi, thanks for your response. I have downloaded the model. I use my ImageNet dataset to test caffe model zoo, and the result is reasonable.
qun.luo@cuda2:/data/lcm/data/imagenet/train$ ls -l ~/.torch/models/
total 145969
-rw------- 1 qun.luo hpc  46827520 May 22 19:34 resnet18-5c106cde.pth
-rw------- 1 qun.luo hpc 102502400 May 22 15:46 resnet50-19c8e357.pth

I use ll to get my imagenet dataset information, could you confirm if it is correct？
qun.luo@cuda2:/data/lcm/data/imagenet$ ls -l val/ILSVRC2012_img_val/ | head 
total 6863808
-rw-r--r-- 1 qun.luo hpc  109527 Jun 13  2012 ILSVRC2012_val_00000001.JPEG
-rw-r--r-- 1 qun.luo hpc  140296 Jun 13  2012 ILSVRC2012_val_00000002.JPEG
-rw-r--r-- 1 qun.luo hpc  122660 Jun 13  2012 ILSVRC2012_val_00000003.JPEG
-rw-r--r-- 1 qun.luo hpc   84885 Jun 13  2012 ILSVRC2012_val_00000004.JPEG
-rw-r--r-- 1 qun.luo hpc  130340 Jun 13  2012 ILSVRC2012_val_00000005.JPEG
-rw-r--r-- 1 qun.luo hpc  151397 Jun 13  2012 ILSVRC2012_val_00000006.JPEG
-rw-r--r-- 1 qun.luo hpc  165863 Jun 13  2012 ILSVRC2012_val_00000007.JPEG
-rw-r--r-- 1 qun.luo hpc  107423 Jun 13  2012 ILSVRC2012_val_00000008.JPEG
-rw-r--r-- 1 qun.luo hpc  114708 Jun 13  2012 ILSVRC2012_val_00000009.JPEG

**Comment by nzmora** [comment]
Hi @qunluo,

Your ImageNet (ILSVRC2012) directory structure seems to be wrong.  It should look like this: https://github.com/pytorch/examples/issues/275#issuecomment-358819144.

```
train/
  ├── n02119789
  │   ├── n02119789_10007.JPEG
  │   ├── n02119789_10011.JPEG
  │   ├── ......
  ├── ......
  val/
  ├── n02119789
  │   ├── ILSVRC2012_val_00000756.JPEG
  │   ├── ILSVRC2012_val_00001260.JPEG
  │   ├── ......
  ├── ......
```

Files are ordered by their synsets (synonym sets) which are represented by WordNet IDs.  WordNet is a lexical database of English.
In the example above, the ID is n02119789 which can be browsed [here](http://image-net.org/synset?wnid=n02119789) and it represents the ```kit fox, Vulpes macrotis``` class.

FB research [provide instructions and scripts](https://github.com/facebook/fb.resnet.torch/blob/master/INSTALL.md#download-the-imagenet-dataset) to create the ImageNet directory structure that the PyTorch Imagenet dataloader expects.

For completeness: class IDs can be found [here](https://gist.github.com/yrevar/942d3a0ac09ec9e5eb3a); you can browse synsets [here](http://image-net.org/challenges/LSVRC/2012/browse-synsets).

Cheers
Neta

**Comment by qunluo** [comment]
Thanks @nzmora. I have got the same accuracy described in command_line.md.

**Comment by HaFred** [comment]
> Hi @qunluo,
> 
> Your ImageNet (ILSVRC2012) directory structure seems to be wrong. It should look like this: [pytorch/examples#275 (comment)](https://github.com/pytorch/examples/issues/275#issuecomment-358819144).
> 
> ```
> train/
>   ├── n02119789
>   │   ├── n02119789_10007.JPEG
>   │   ├── n02119789_10011.JPEG
>   │   ├── ......
>   ├── ......
> val/
>   ├── n02119789
>   │   ├── ILSVRC2012_val_00000756.JPEG
>   │   ├── ILSVRC2012_val_00001260.JPEG
>   │   ├── ......
>   ├── ......
> ```
> 
> Files are ordered by their synsets (synonym sets) which are represented by WordNet IDs. WordNet is a lexical database of English.
> In the example above, the ID is n02119789 which can be browsed [here](http://image-net.org/synset?wnid=n02119789) and it represents the `kit fox, Vulpes macrotis` class.
> 
> FB research [provide instructions and scripts](https://github.com/facebook/fb.resnet.torch/blob/master/INSTALL.md#download-the-imagenet-dataset) to create the ImageNet directory structure that the PyTorch Imagenet dataloader expects.
> 
> For completeness: class IDs can be found [here](https://gist.github.com/yrevar/942d3a0ac09ec9e5eb3a); you can browse synsets [here](http://image-net.org/challenges/LSVRC/2012/browse-synsets).
> 
> Cheers
> Neta

Hi @nzmora , sorry to bother. But it seems to me in the [FB link](https://github.com/facebook/fb.resnet.torch/blob/master/INSTALL.md#download-the-imagenet-dataset) your provided, the `valprep.sh` seems incomplete. It only deals with 50k val images, but in the val.tar there are 100k images. I wonder how to mv the remaining 50k? Thank you very much.
